### ISSUE TITLE
Java Client should not return null for 404 responses

### ISSUE THREAD

**Issue by steven-barnes** [killbill/killbill-client-java#135]
For code like:
```java
subscriptionApi.getSubscriptionByKey("xyzzy", requestOptions)
```

The client returns null on 404 errors. This is unexpected. The client should throw a KillBillClientException containing the response code.

**Comment by pierre** [comment]
Yes, we've talked about enhancing this in the past. This behaviour has been in place for 12+ years though, so it'd be a large breaking change.
